### DISCUSSION TITLE
2022-01-05 - monthly meeting

### DISCUSSION THREAD

**Discussion by explodecomputer** [MRCIEU/opengwas#13] [Meeting notes]
Agenda

- Welcome to Jason (and Peter to these meetings)
- Gib updates from Boston visit
  - Using buckets to host data for API queries unlikely to be particularly fast but will be free. Worth exploring?
  - Interest from HAIL - especially in thinking about storing study specific LD info
- Elasticsearch updates
  - Need list of finn-b datasets that have successfully uploaded
  - Need to reindex those that didn't
  - Need to change alias for finn-b
  - AOB
- Meta data errors - Phil has gone through everything and provided info here: https://github.com/MRCIEU/opengwas/issues/12, in particular, it looks like the effect allele column may have been mis-specified for the ukb-e batch. 
- Ontology mapping plan: https://github.com/MRCIEU/opengwas/discussions/7
- Discussion about license / terms of use - https://uob-my.sharepoint.com/:w:/g/personal/gh13047_bristol_ac_uk/EeoyEMHe-oROh_SDveLktMIBAh_L7JjE6JERmAOTZZoxKg?e=8IAnZC
- AOB

**Comment by explodecomputer** [comment]
Think about moving database to tiered system

- Most needed datasets to elasticsearch
- Lower priority datasets in buckets
- Potentially move everything to haildb format - if they could provide effective indexing for row, column and row/column searches

Possibilities
- Hail
- GenomicsDB
- Parquet
- TileDB

Potential funding for software developers
- Elixir
- CZI 

Ask RED for advice on funding

---

Set up meeting with Jason to discuss elasticsearch db

---

When metadata is incorrect e.g. incorrect alleles etc, don't necessarily need to delete elasticsearch data, just need to set qc to fail for that ID and upload a new dataset.

Create an endpoint for replacing datasets
1. Use existing flat files to upload to a new ID
2. Hide old ID by setting QC to fail and providing a note of where to find new version
3. Update / create qc json file that needs to be rsynced to flat file locations

@mcgml @mightyphil2000 does this plan look ok or did I misinterpret anything?

---

Tom P to look at getting R packages ready for bioconductor / CRAN and to join next meeting with Rob and Vince

**Comment by mightyphil2000** [comment]
@explodecomputer, as I mentioned at our last meeting, I found quite a few EBI datasets that seem to have the incorrect effect allele specified. I've done some further digging and I think the problem is on our side, rather the EBI side. For example, rs174546, effect allele T, is associated with higher LDL cholesterol (beta=0.05) in Open GWAS (ebi-a-GCST000759) but in the GWAS catalog allele T is associated with lower LDL cholesterol across multiple studies. I also downloaded dataset GCST000759 from the EBI FTP site and confirmed that rs174546 T is inversely associated with LDL cholesterol (beta=-0.05). http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000759/harmonised/

out<- ieugwasr::associations(id= "ebi-a-GCST000759", variants="rs174546",proxies=0)
CheckSumStats::gwas_catalog<-gwas_catalog_hits2(trait="LDL cholesterol")
gwas_catalog[gwas_catalog$study_id=="GCST000759" & gwas_catalog$rsid == "rs174546" ,]
